python-lib/results/ab_statistics.py

```diff
@@ -0,0 +1,52 @@
+import pandas as pd
+
+from constants import Group, Columns
+
+
+class AbStatistics(object):
+    def __init__(self, user_reference_column: str, group_column: str, conversion_column: str):
+        self.user_reference_column = user_reference_column
+        self.group_column = group_column
+        self.conversion_column = conversion_column
+
+    def compute(self, results_df):
+        results_df = self.check_results_df(results_df)
+        aggregation = self.aggregate_by_group(results_df)
+        statistics_df = self.format_statistics(aggregation)
+        return statistics_df
+
+    def check_results_df(self, result_df):
+        valid_rows = result_df[[self.user_reference_column, self.group_column, self.conversion_column]].dropna()
+        not_empty_rows_nb = valid_rows.shape[0]
+        if not result_df[self.user_reference_column].is_unique:
+            raise ValueError("There should be only one row per user in the input dataset")
+        if not_empty_rows_nb < 2:
+            raise ValueError("The input dataset should contain at least 2 users with conversion and group references")
+        invalid_groups = valid_rows[(valid_rows[self.group_column] != Group.A.value) & (valid_rows[self.group_column] != Group.B.value)]
```

Comment:
I feel like this is too strict as constraint. We can simply check if the group indicator is binary, what do you think ?

Reply:
yes, would be more flexible ! 
I need to adapt the web app accordingly

The idea was also that the user is supposed to perform a join before using this recipe. But I guess it's a big constraint

Reply:
I tried to use a custom field to choose the values for A and B (using a Python script and a do function) but it does not work so well as it is not dynamic. So you need to refresh the page to get the values. I guess to do so, we would need a custom angular form. So right now, your solution seems best

Reply:
done in https://github.com/dataiku/dss-plugin-ab-testing/pull/2/commits/d3b6ebc0487f2213eef0edcfd63fd6bba5e990b4